src/status_im/utils/mixpanel.cljs

```diff
@@ -1,38 +1,81 @@
 (ns status-im.utils.mixpanel
   (:require-macros [status-im.utils.slurp :as slurp])
-  (:require [cljs.reader :as reader]
+  (:require [cljs.core.async :as async]
+            [cljs.reader :as reader]
             [goog.crypt.base64 :as b64]
             [status-im.utils.build :as build]
             [status-im.utils.config :as config]
             [status-im.utils.datetime :as datetime]
             [status-im.utils.http :as http]
             [status-im.utils.platform :as platform]
-            [status-im.utils.types :as types]))
+            [status-im.utils.types :as types]
+            [taoensso.timbre :as log]))
 
 (def base-url "http://api.mixpanel.com/")
 (def base-track-url (str base-url "track/"))
 
 (defn encode [m]
   (b64/encodeString (types/clj->json m)))
 
-(defn- build-url [id label props]
-  (str base-track-url
-       "?data="
-       (encode {:event      label
-                :properties (merge
-                              {:token       config/mixpanel-token
-                               :distinct_id id
-                               :os          platform/os
-                               :os-version  platform/version
-                               :app-version build/version
-                               :time        (datetime/timestamp)}
-                              props)})))
-
-(defn track [id label props]
-  (http/get (build-url id label props) nil identity))
+(defn- make-event [id label props]
+  {:event      label
+   :properties (merge
+                {:token       config/mixpanel-token
+                 :distinct_id id
+                 :os          platform/os
+                 :os-version  platform/version
+                 :app-version build/version
+                 :time        (datetime/timestamp)}
+                props)})
 
-;; Mixpanel events definition
+;; holds events that are accumulated while offline. will start throwing away old
+;; events if we accumulate more than 2000
+(def ^:private pending-events-queue (async/chan (async/sliding-buffer 2000)))
+
+(defn- submit-batch
+  "Submit a batch of events to mixpanel via POST"
+  [events]
+  (let [done-chan (async/chan)]
+    (log/debug "submitting" (count events) "events")
+    (http/post base-track-url
+               (str "data=" (encode (into [] events)))
+               (fn [_]
+                 (log/debug "successfully submitted events")
+                 (async/go (async/close! done-chan)))
+               (fn [error]
+                 (log/error "error while submitting events" error)
```

Comment:
Are you sure `log/error` doesn't bring react-native "redbox" error to the user? Not sure user wants to know about such error.